homeassistant/components/pvoutput/sensor.py

```diff
@@ -125,7 +126,18 @@ async def async_added_to_hass(self):
     def _async_update_from_rest_data(self):
         """Update state from the rest data."""
         try:
-            self.pvcoutput = self.status._make(self.rest.data.split(","))
+            raw_data = self.rest.data.split(",")
+            # ["20210615", "21:20", "23608", "24", "NaN", "NaN", "0.007", "56.5", "242.0"]
+            # If the retrieved datetime is older than 5 minutes, set the power to 0.
+            reading_datetime = datetime.datetime.strptime(
+                raw_data[0] + raw_data[1], "%Y%m%d%H:%M"
+            )
+            if datetime.datetime.now() > reading_datetime + timedelta(minutes=5):
+                _LOGGER.debug(
+                    "Collected data was older than 5 minutes, so set the power to 0"
+                )
+                raw_data[3] = "NaN"
```

Comment:
Please extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices

Reply:
Hey @MartinHjelmare - OK thanks for the feedback. I'll probably not be able to get to this for some time as will be a much bigger task than the teeny bugfix implemented to the current code to stop it from displaying wrong data.
I'll volume map over the core file for now on my running instance which is solving my issue for now.

Thanks again.